microsoft-edge/devtools-guide-chromium/whats-new/2024/01/devtools-121.md

```diff
@@ -0,0 +1,121 @@
+---
+title: What's New in DevTools (Microsoft Edge 121)
+description: "Z-index and DOM tabs in 3D View tool are now working. Correct Hide debugger or Show debugger icon is shown in Sources tool. And more."
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: devtools
+ms.date: 01/25/2024
+---
+# What's New in DevTools (Microsoft Edge 121)
+
+[!INCLUDE [Microsoft Edge team note for top of What's New](../../includes/edge-whats-new-note.md)]
+
+
+<!-- ====================================================================== -->
+## Z-index and DOM tabs in 3D View tool are now working
+
+<!-- Subtitle: In recent versions of Microsoft Edge, the Z-index and DOM tabs weren't working. In Microsoft Edge 121, this issue has been resolved. -->
+
+In recent versions of Microsoft Edge, there was a bug in the **3D View** tool that caused the **Z-index** and **DOM** tabs to not work correctly.  The tabs would crash or freeze and show the content for the **Composited Layers** tab instead.  This issue has been fixed in Microsoft Edge 121:
+
+![The Z-index tab in the 3D View tool](./devtools-121-images/z-index-tab.png)
+
+See also:
+* [Navigate webpage layers, z-index, and DOM using the 3D View tool](../../../3d-view/index.md)
+
+
+<!-- ====================================================================== -->
+## Correct Hide debugger or Show debugger icon is shown in Sources tool
+
+<!-- Subtitle: In previous versions of Microsoft Edge, the Hide debugger icon was displayed instead of the Show debugger icon and vice versa in the Sources tool. In Microsoft Edge 121, this issue has been resolved. -->
+
+The icon used to collapse or expand the **Debugger** pane on the right in the **Sources** tool was incorrect.  The **Hide debugger** icon was displayed instead of the **Show debugger** icon, and vice versa:
+
+![Icon in the Sources tool, before](./devtools-121-images/open-close-button-sources-before.png)
+
+In Microsoft Edge 121, the correct icon is now shown:
+
+![Icon in the Sources tool, after](./devtools-121-images/open-close-button-sources-after.png)
+
+See also:
+* [Sources tool overview](../../../sources/index.md)
+
+
+<!-- ====================================================================== -->
+## Added source map support in the Coverage tool
+
+<!-- Subtitle: Quickly identify and eliminate unused code with the improved source map support in the Coverage tool. -->
+
+Before Microsoft Edge 121, the **Coverage** tool only displayed information for generated files, and omitted source files even when a source map was attached.  The **Coverage** tool has now been improved so that when a generated file has a source map attached, the **Coverage** tool displays information for the source file as well as for the generated file.
+
```

Comment:
need screenshot

Reply:
You can create a before/after screenshot by using https://microsoftedge.github.io/Demos/slow-calendar/public/
It's a recent demo I've been working on that uses source maps.

Steps to take screenshots:

1. Open the page and DevTools.
1. In DevTools, use the command menu, and run the **Show Coverage** command.
1. In the **Coverage** tool, click the **Refresh** button.

Before 121, the coverage report should contain a line for the `bundle.js` file.
Since 121, the `bundle.js` line in the coverage report can be expanded to show all of the original source files that make up this compiled bundle file:

![image](https://github.com/MicrosoftDocs/edge-developer/assets/1152698/fd49f795-02d2-4554-a1f6-90b131597607)

Reply:
added final png